pyanaconda/modules/storage/nvme/nvme_interface.py

```diff
@@ -0,0 +1,35 @@
+#
+# DBus interface for the NVMe module.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from dasbus.server.interface import dbus_interface
+from dasbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base import KickstartModuleInterfaceTemplate
+from pyanaconda.modules.common.constants.objects import NVME
+
+
+@dbus_interface(NVME.interface_name)
+class NVMEInterface(KickstartModuleInterfaceTemplate):
+    """DBus interface for the NVMe module."""
+
+    def WriteConfiguration(self):
+        """Write the configuration to sysroot.
+
+        FIXME: This is just a temporary method.
+        """
```

Comment:
What is the plan there, then?

Reply:
We already talked about this. The method is here to be consistent with other submodules of this type. It is a left-over workaround from the modularization. The method is used only by the Storage module, so the DBus API can be eventually dropped and the submodules handled internally. However, that's too much work and not really relevant to this issue.

Reply:
I have no idea, the same `FIXME` is in the iSCSI interface (where I copied it from) and in FCoE and ZFCP too.